### PR TITLE
Add links to HttpContext.[Verb]Async in authn overview

### LINKED ISSUE #17186: Add links to e.g. HttpContext.AuthenticateAsync references
There are three places in the topic that reference `HttpContext.[Verb]Async`:

- See `HttpContext.AuthenticateAsync`.
- See `HttpContext.ChallengeAsync`.
- See `HttpContext.ForbidAsync`.

None of these are links, but I'd expect them to take me to e.g. [ChallengeAsync](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.authenticationhttpcontextextensions.challengeasync?view=aspnetcore-3.1).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 393b3379-6061-50a4-0975-b3d1d82b660b
* Version Independent ID: 48e4e204-ea22-c225-0cbc-0d811bdb90ed
* Content: [Overview of ASP.NET Core Authentication](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/?view=aspnetcore-3.1)
* Content Source: [aspnetcore/security/authentication/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/security/authentication/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @mjrousos
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by serpent5**
Fixes #17186.

### CODE REVIEW

File: aspnetcore/security/authentication/index.md [hunk 1]
```diff
@@ -74,14 +74,14 @@ Based on the authentication scheme's configuration and the incoming request cont
 
 ### Authenticate
 
-An authentication scheme's authenticate action is responsible for constructing the user's identity based on request context. It returns an <xref:Microsoft.AspNetCore.Authentication.AuthenticateResult> indicating whether authentication was successful and, if so, the user's identity in an authentication ticket. See `HttpContext.AuthenticateAsync`. Authenticate examples include:
+An authentication scheme's authenticate action is responsible for constructing the user's identity based on request context. It returns an <xref:Microsoft.AspNetCore.Authentication.AuthenticateResult> indicating whether authentication was successful and, if so, the user's identity in an authentication ticket. See [HttpContext.AuthenticateAsync](xref:Microsoft.AspNetCore.Authentication.AuthenticationHttpContextExtensions.AuthenticateAsync*). Authenticate examples include:
```
Comment (scottaddie) [inline aspnetcore/security/authentication/index.md]:
```suggestion
An authentication scheme's authenticate action is responsible for constructing the user's identity based on request context. It returns an <xref:Microsoft.AspNetCore.Authentication.AuthenticateResult> indicating whether authentication was successful and, if so, the user's identity in an authentication ticket. See <xref:Microsoft.AspNetCore.Authentication.AuthenticationHttpContextExtensions.AuthenticateAsync%2A>. Authenticate examples include:
```

File: aspnetcore/security/authentication/index.md [hunk 2]
```diff
@@ -74,14 +74,14 @@ Based on the authentication scheme's configuration and the incoming request cont
...
 * A JWT bearer scheme deserializing and validating a JWT bearer token to construct the user's identity.
 
 ### Challenge
 
-An authentication challenge is invoked by Authorization when an unauthenticated user requests an endpoint that requires authentication. An authentication challenge is issued, for example, when an anonymous user requests a restricted resource or clicks on a login link. Authorization invokes a challenge using the specified authentication scheme(s), or the default if none is specified. See `HttpContext.ChallengeAsync`. Authentication challenge examples include:
+An authentication challenge is invoked by Authorization when an unauthenticated user requests an endpoint that requires authentication. An authentication challenge is issued, for example, when an anonymous user requests a restricted resource or clicks on a login link. Authorization invokes a challenge using the specified authentication scheme(s), or the default if none is specified. See [HttpContext.ChallengeAsync](xref:Microsoft.AspNetCore.Authentication.AuthenticationHttpContextExtensions.ChallengeAsync*). Authentication challenge examples include:
```
Comment (scottaddie) [inline aspnetcore/security/authentication/index.md]:
```suggestion
An authentication challenge is invoked by Authorization when an unauthenticated user requests an endpoint that requires authentication. An authentication challenge is issued, for example, when an anonymous user requests a restricted resource or clicks on a login link. Authorization invokes a challenge using the specified authentication scheme(s), or the default if none is specified. See <xref:Microsoft.AspNetCore.Authentication.AuthenticationHttpContextExtensions.ChallengeAsync%2A>. Authentication challenge examples include:
```

File: aspnetcore/security/authentication/index.md [hunk 3]
```diff
@@ -90,7 +90,7 @@ A challenge action should let the user know what authentication mechanism to use
 
 ### Forbid
 
-An authentication scheme's forbid action is called by Authorization when an authenticated user attempts to access a resource they are not permitted to access. See `HttpContext.ForbidAsync`. Authentication forbid examples include:
+An authentication scheme's forbid action is called by Authorization when an authenticated user attempts to access a resource they are not permitted to access. See [HttpContext.ForbidAsync](xref:Microsoft.AspNetCore.Authentication.AuthenticationHttpContextExtensions.ForbidAsync*). Authentication forbid examples include:
```
Comment (scottaddie) [inline aspnetcore/security/authentication/index.md]:
```suggestion
An authentication scheme's forbid action is called by Authorization when an authenticated user attempts to access a resource they are not permitted to access. See <xref:Microsoft.AspNetCore.Authentication.AuthenticationHttpContextExtensions.ForbidAsync%2A>. Authentication forbid examples include:
```

**Comment by scottaddie** [comment]
@serpent5 What you had will certainly work, but we now prefer to use the API ref link format that I've suggested.

**Comment by serpent5** [comment]
@scottaddie Thanks for the info. Do you know if I can test these links locally? They don't resolve when using docfx, but I feel like there was a time when they did.

**Comment by scottaddie** [comment]
@serpent5 The stable version of the DocFx CLI won't work right now because we've transitioned to v3 of DocFx: https://github.com/dotnet/AspNetCore.Docs/blob/a510489773756c7fce7ccd9c1672348b9bd264fc/.openpublishing.publish.config.json#L65. Until that's fixed, I can't think of a great solution to preview the links.

**Comment by scottaddie** [comment]
Thank you for your help, @serpent5!

**Pull request merged**